lib/pure/strformat.nim

```diff
@@ -258,7 +258,9 @@ template callFormat(res, arg) {.dirty.} =
     # workaround in order to circumvent 'strutils.format' which matches
     # too but doesn't adhere to our protocol.
     res.add arg
-  elif compiles(format(arg, res)):
+  elif compiles(format(arg, res)) and
```

Comment:
Maybe the following would be better?

```nim
elif compiles((var res = ""; format(arg, res)))
```

Strange that the current code doesn't catch the "return value needs to be discarded" error. Maybe the above version will?